numpy/core/src/multiarray/conversion_utils.c

```diff
@@ -553,6 +557,13 @@ PyArray_OrderConverter(PyObject *object, NPY_ORDER *val)
     }
     else {
         str = PyBytes_AS_STRING(object);
+        if (strlen(str) != 1) {
+            DEPRECATE("Non length-one string passed in " \
```

Comment:
`Deprecate/PyErr_WarnEx` can return an error value (specifically, this happens if you do for example `import warnings; warnings.simplefilter("error", DeprecationWarning)`.
In that case you need to check for a negative return, do the typical cleanup (which might be nothing in this case) and return an error return (NULL or -1 normally).